### ISSUE TITLE
Debugger upon breakpoint or step not loading source path and not highlighting line unless clicking on the call stack item

### ISSUE THREAD

**Issue by bernhardttom** [microsoft/vscode#107187]
VSCode Version: Version: 1.49.1 (system setup)
Commit: 58bb7b2331731bf72587010e943852e13e6fd3cf
Date: 2020-09-16T23:27:51.792Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19041

Steps to Reproduce:

1. Write a VSCode extension for a language and specify "debuggers"
2. Write a debug adapter
3. Set a breakpoint and launch
4. Expected VSCode to load the file that is at the top of the callstack and to position to line, but it only does when clicking on callstack item
5. When stepping, expected VSCode to position to next line, but it only does when clicking on callstack item

I have an SQL-based language and we are developing a debugger. The debug adapter returns a [Source ](https://microsoft.github.io/debug-adapter-protocol/specification#Types_Source) that points to a Java file. The debug adapter returns a stack frame with line numbers. However VSCode only loads the file when actually clicking on the callstack which makes stepping impossible.

Do I need to register a text document content provider and use my own scheme?

**Comment by bernhardttom** [comment]
Solved.

The problem was the thread-id field provided by the debug adapter which wasn't the same value between "threads()" and the ""stopped" event.

We found this out by getting a trace of the debug adapter request and responses and inspecting carefully.

For the sake of helping others, here is the code to see the debug adapter requests and responses:
`	vscode.debug.registerDebugAdapterTrackerFactory('*', {
		createDebugAdapterTracker(session: vscode.DebugSession) {
		  return {
			onWillReceiveMessage: m => console.log(`> ${JSON.stringify(m, undefined, 2)}`),
			onDidSendMessage: m => console.log(`< ${JSON.stringify(m, undefined, 2)}`)
		  };
		}
	  });
`
